pkg/apis/resource/validation/validation.go

```diff
@@ -216,23 +216,29 @@ func validateExactDeviceRequest(request resource.ExactDeviceRequest, fldPath *fi
 	var allErrs field.ErrorList
 	allErrs = append(allErrs, validateDeviceClass(request.DeviceClassName, fldPath.Child("deviceClassName"))...)
 	allErrs = append(allErrs, validateSelectorSlice(request.Selectors, fldPath.Child("selectors"), stored)...)
-	allErrs = append(allErrs, validateDeviceAllocationMode(request.AllocationMode, request.Count, fldPath.Child("allocationMode"), fldPath.Child("count"))...)
+	allErrs = append(allErrs, validateDeviceAllocationMode(request.AllocationMode, request.Count, fldPath.Child("allocationMode"), fldPath.Child("count"), false /* subRequest */)...)
 	for i, toleration := range request.Tolerations {
 		allErrs = append(allErrs, validateDeviceToleration(toleration, fldPath.Child("tolerations").Index(i))...)
 	}
 	return allErrs
 }
 
-func validateDeviceAllocationMode(deviceAllocationMode resource.DeviceAllocationMode, count int64, allocModeFldPath, countFldPath *field.Path) field.ErrorList {
+func validateDeviceAllocationMode(deviceAllocationMode resource.DeviceAllocationMode, count int64, allocModeFldPath, countFldPath *field.Path, subRequest bool) field.ErrorList {
 	var allErrs field.ErrorList
 	switch deviceAllocationMode {
 	case resource.DeviceAllocationModeAll:
 		if count != 0 {
 			allErrs = append(allErrs, field.Invalid(countFldPath, count, fmt.Sprintf("must not be specified when allocationMode is '%s'", deviceAllocationMode)))
 		}
 	case resource.DeviceAllocationModeExactCount:
-		if count <= 0 {
-			allErrs = append(allErrs, field.Invalid(countFldPath, count, "must be greater than zero"))
+		if subRequest {
+			if count < 0 {
+				allErrs = append(allErrs, field.Invalid(countFldPath, count, "must be zero or greater"))
```

Comment:
do we need to handle two distinct error conditions here? can we collapse this into

```
if count <= 0 {
    allErrs = append(allErrs, field.Invalid(countFldPath, count, "must be zero or greater"))
}